src/test/rustdoc-ui/intra-link-errors.stderr

```diff
@@ -0,0 +1,93 @@
+error: unresolved link to `path::to::nonexistent::module`
+  --> $DIR/intra-link-errors.rs:8:6
+   |
+LL | /// [path::to::nonexistent::module]
+   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |
+note: the lint level is defined here
+  --> $DIR/intra-link-errors.rs:1:9
+   |
+LL | #![deny(broken_intra_doc_links)]
+   |         ^^^^^^^^^^^^^^^^^^^^^^
+   = note: no item named `path::to` is in scope
+   = help: to escape `[` and `]` characters, add '\' before them like `\[` or `\]`
+
+error: unresolved link to `f::A`
+  --> $DIR/intra-link-errors.rs:14:6
+   |
+LL | /// [f::A]
+   |      ^^^^
+   |
+   = note: this link partially resolves to the function `f`
+   = note: `f` is a function, not a module or type, and cannot have associated items
+
+error: unresolved link to `S::A`
+  --> $DIR/intra-link-errors.rs:19:6
+   |
+LL | /// [S::A]
+   |      ^^^^
+   |
+   = note: this link partially resolves to the struct `S`
+   = note: no `A` in `S`
```

Comment:
A rustacean on discord suggested that this be consistent with the more verbose form for primitives:
```suggestion
error: unresolved link to `S::A`
  --> $DIR/intra-link-errors.rs:19:6
   |
LL | /// [S::A]
   |      ^^^^
   |
   = note: this link partially resolves to the struct `S`
   = note: `S` does not have a field or associated item named `fmt`
```